### PR TITLE
chore(enos): Move `terraform-enos-aws-infra` to `boundary`

### PR DESCRIPTION
**Pull request by moduli**
The terraform-enos-aws-infra repo was a terraform module to set up the base aws infrastructure. Vault made some local updates that might mitigate flakiness we've been seeing ("can't delete due to dependencies" error). We are replacing our usage of that module with Vault's local copy.

### CODE REVIEW

File: enos/modules/aws_vpc/main.tf [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  default     = ["amd64", "arm64"]
+}
+
+locals {
+  // AWS AMIs standardized on the x86_64 label for 64bit x86 architectures, therefore amd64 should be rather x86_64.
+  architecture_filters = [for arch in var.ami_architectures : (arch == "amd64" ? "x86_64" : arch)]
+  common_tags = merge(
+    var.common_tags,
+    {
+      "Module" = "terraform-enos-aws-infra" // TODO:  Rename this once terraform-enos-aws-boundary is moved
```
Comment (moduli) [inline enos/modules/aws_vpc/main.tf:52]: Right now, there's a dependency with `terraform-enos-aws-boundary` which looks for AWS items with the tag. I will rename this once I move `terraform-enos-aws-boundary` into the `boundary` repo.

File: enos/modules/aws_vpc/main.tf
Comment (moduli) [inline enos/modules/aws_vpc/main.tf:1]: This was mostly taken from https://github.com/hashicorp/vault/blob/main/enos/modules/create_vpc/main.tf with some slight modifications to `variables` and `outputs` to match what boundary was using from https://github.com/hashicorp/terraform-enos-aws-infra

File: enos/ci/service-user-iam/main.tf [hunk 2]
```diff
@@ -80,6 +80,7 @@ data "aws_iam_policy_document" "enos_policy_document" {
       "ec2:CreateVPC",
       "ec2:DeleteInternetGateway",
       "ec2:DeleteKeyPair",
+      "ec2:DeleteRoute",
```
Comment (moduli) [inline enos/ci/service-user-iam/main.tf:83]: Needed to give permissions to the CI service user to delete routes.

**Review by johanbrandhorst — APPROVED** [review]

**Comment by moduli** [comment]
Looks like there's one issue to figure out. One scenario is failing because of one enos module not being able to find the aws route. Have a lead that I'm working on

**Pull request merged**